### PR TITLE
common, msgr/async: bufferlist::claim_append() doesn't require intermediaries anymore

### PR DESCRIPTION
**Pull request by rzarzynski**
This a follow-up to the #35078.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/include/buffer.h [hunk 1]
```diff
@@ -1066,6 +1066,9 @@ struct error_code;
 
     void claim(list& bl);
     void claim_append(list& bl);
```
Comment (rzarzynski) [inline src/include/buffer.h:1068]: The entire `claim_...` family deserves deprecation, I believe. `claim_append()` is basically pre-C++11 way to say `dst.append(std::move(src))`.

Comment (idryomov) [reply src/include/buffer.h:1068]: Yeah, that's pretty much what I thought and that's the reason I didn't include this trivial change in the msgr2.1 PR -- I assumed you'd want to sweep the codebase and get rid of `claim()` and `claim_append()`.

File: src/msg/async/frames_v2.cc [hunk 2]
```diff
@@ -396,7 +394,7 @@ void FrameAssembler::disasm_first_secure_rev1(bufferlist& preamble_bl,
     segment_bl.swap(tmp);
     preamble_bl.splice(sizeof(preamble_block_t), FRAME_PREAMBLE_INLINE_SIZE,
                        &segment_bl);
-    segment_bl.claim_append(tmp);
+    segment_bl.claim_append(std::move(tmp));
```
Comment (idryomov) [inline src/msg/async/frames_v2.cc:397]: This change seems necessary.

File: src/include/buffer.h (hunk 1 above)
Comment (tchaikov) [reply src/include/buffer.h:1068]: @rzarzynski do you plan to mark `claim_append()` and friends with `[[deprecated]]` specifier in a follow-up PR?

File: src/include/buffer.h [hunk 3]
```diff
@@ -1066,6 +1066,9 @@ struct error_code;
...
+    void claim_append(list&& bl) {
```
Comment (tchaikov) [inline src/include/buffer.h:1069]: why not just add
```c++
void append(list&& bl)
```
?

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2020-06-23_07:49:50-rados-wip-kefu-testing-2020-06-23-1007-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-06-23_15:15:59-rados-wip-kefu-testing-2020-06-23-1353-distro-basic-smithi/

---

- http://pulpito.ceph.com/kchai-2020-06-23_07:49:50-rados-wip-kefu-testing-2020-06-23-1007-distro-basic-smithi/5171960, no clues, osd.2 failed to reply queries from osd.0 when peering. but there was no clues in its log.
- http://pulpito.ceph.com/kchai-2020-06-23_15:15:59-rados-wip-kefu-testing-2020-06-23-1353-distro-basic-smithi/5173648:  tracked by https://tracker.ceph.com/issues/45721
- http://pulpito.ceph.com/kchai-2020-06-23_15:15:59-rados-wip-kefu-testing-2020-06-23-1353-distro-basic-smithi/5173654/: no tests failed..

**Comment by rzarzynski** [comment]
@tchaikov: yup, I want to deprecate `claim_append` and move everything to `append(list&&)` in a much bigger changeset.

**Pull request merged**